stylix/hm/icon.nix

```diff
@@ -0,0 +1,22 @@
+{ config, lib, pkgs, ... }:
+
+let
+	cfg = config.stylix.iconTheme;
+	inherit (config.stylix) polarity;
+in {
+	imports = [ ../icon.nix ];
+	config = lib.mkIf (config.stylix.enable && cfg.enable && pkgs.stdenv.hostPlatform.isLinux) {
+		gtk = {
+			iconTheme = {
+				inherit (cfg) package;
+				name = builtins.head (lib.filter (x: null != x) [
+				  ({
+					inherit (cfg) dark light;
+				  }."${polarity}" or null)
+				  cfg.dark
+				  cfg.light
+				]);
```

Comment:
This is very clever. However, it might be better to prioritize readability with simple if-statements.

Reply:
Initially I did this as an if statement in my head, but it came out very large and fairly confusing. This code does require some knowledge about functional programing, but it comes out much more simpler if you do know them.
For reference this statement unfolds in to 3 if statements, one of which nests an additional if statement. If you insist I can attempt to rewrite it though.

Reply:
> For reference this statement unfolds in to 3 if statements, one of which nests an additional if statement.

The nested if-statement, which is probably the most confusing part could be flattened to something like:

```nix
if polarity == "dark"
then dark
else if polarity == "light"
then light
```

> If you insist I can attempt to rewrite it though.

Just send it as a comment here before pushing it. Seeing both makes it easier to compare which one is better.

Reply:
Something like this.
```nix
name = if cfg.light != null && cfg.dark != null then
	if polarity == "dark"
	then cfg.dark
	else if polarity == "light"
	then cfg.light
else if cfg.light == null 
then cfg.dark
else if cfg.dark == null 
then cfg.light;
```

Reply:
Also I'm not entirely sure how an `else` should look like in here.

Reply:
Should we not simplify the `name` to:

```nix
name = "stylix";
```

Many Stylix themes are called `Stylix` or `stylix`.

Reply:
No because this field is the name of the iconTheme, so based on what is going to be here the iconTheme will be selected. I implemented the same interface as someone in the #458 suggested.

Reply:
@trueNAHO Is there any other questions regarding the code, or are we just waiting for danth to approve of this?

Reply:
> Is there any other questions regarding the code

Considering the scope of this icon implementation it might be good to get @danth's opinion and approval.

> or are we just waiting for danth to approve of this?

Since the current implementation is fine for me, we can just leave it like this and move forward.